src/main/scala/com/microsoft/sqlserver/jdbc/spark/utils/BulkCopyUtils.scala

```diff
@@ -239,53 +267,85 @@ object BulkCopyUtils extends Logging {
         val dfCols = df.schema
 
         val tableCols = getSchema(rs, JdbcDialects.get(url))
+        val computedCols = getComputedCols(conn, dbtable)
+        var dfComputedCols : List[String] = List()
+
         val prefix = "Spark Dataframe and SQL Server table have differing"
 
-        assertCondition(dfCols.length == tableCols.length,
-            s"${prefix} numbers of columns")
+        if (computedCols.length == 0) {
+            assertCondition(dfCols.length == tableCols.length,
+                s"${prefix} numbers of columns")
+        }
+        else {
+            val dfColNames =  df.schema.fieldNames.toList
+            var colCount = 0
+            for (j <- 0 to computedCols.length-1){
+                if (isCaseSensitive && dfColNames.contains(computedCols(j)) ||
+                  !isCaseSensitive && dfColCaseMap.contains(computedCols(j).toLowerCase()) && dfColCaseMap(computedCols(j).toLowerCase()) == computedCols(j)) {
+                    colCount += 1
+                    dfComputedCols = dfComputedCols :+ computedCols(j)
+                }
+            }
+            assertCondition(dfCols.length-colCount+computedCols.length == tableCols.length,
+                s"${prefix} numbers of columns")
+        }
 
         val result = new Array[ColumnMetadata](tableCols.length)
 
         for (i <- 0 to tableCols.length-1) {
             val tableColName = tableCols(i).name
-            var dfFieldIndex = 0
-            var dfColName:String = ""
-            if (isCaseSensitive) {
-                dfFieldIndex = dfCols.fieldIndex(tableColName)
-                dfColName = dfCols(dfFieldIndex).name
-                assertCondition(tableColName == dfColName,
-                    s"""${prefix} column names '${tableColName}' and
-                        '${dfColName}' at column index ${i} (case sensitive)""")
-            } else {
-                dfFieldIndex = dfCols.fieldIndex(dfColCaseMap(tableColName.toLowerCase()))
-                dfColName = dfCols(dfFieldIndex).name
-                assertCondition(tableColName.toLowerCase() == dfColName.toLowerCase(),
-                    s"""${prefix} column names '${tableColName}' and
-                        '${dfColName}' at column index ${i} (case insensitive)""")
-            }
+            var dfFieldIndex = -1
+            var isAutoIncrement = false
+            if (computedCols.contains(tableColName)) {
+                isAutoIncrement = true
+                if (dfComputedCols.contains(tableColName)){
+                    // tableColName is a computed col in sql table, but also exist in df
+                    if (isCaseSensitive) {
```

Comment:
do we need to set the dfFieldIndex when the column is computed. We are any way going to ignore the value.

Reply:
My previous approach: 1.Computed Col exists in df, mark dfIndex as is, set isAutoIncrement as True. 2.Computed Col not exist in df, set isAutoIncrement True and dfIndex=-1 implies that column is not present.
Now all computed col have dfIndex=-1 and isAutoIncrement as True.